lua/hbac/command/subcommands.lua

```diff
@@ -0,0 +1,45 @@
+local autocommands = require("hbac.autocommands")
+local hbac_config = require("hbac.setup").opts
+local state = require("hbac.state")
+local utils = require("hbac.utils")
+
+local M = {}
+
+M.close_unpinned = function()
+	local curbufnr = vim.api.nvim_get_current_buf()
+	local buflist = utils.get_buffers()
+	for _, bufnr in ipairs(buflist) do
+		local windows_with_buf = vim.fn.win_findbuf(bufnr)
+		if bufnr ~= curbufnr and not state.is_pinned(bufnr) and #windows_with_buf == 0 then
+			hbac_config.close_command(bufnr)
+		end
+	end
+end
+
+M.toggle_pin = function()
+	local bufnr = vim.api.nvim_get_current_buf()
+	local pinned_state = state.toggle_pin(bufnr) and "pinned" or "unpinned"
+	return bufnr, pinned_state
+end
+
+M.pin_all = function()
+	local buflist = utils.get_buffers()
+	for _, bufnr in ipairs(buflist) do
+		state.pinned_buffers[bufnr] = state.pin_all
+	end
+	state.pin_all = not state.pin_all
+	return state.pin_all
```

Comment:
This probably could work as a module scoped side effect, since it's unlikely other modules would reach for this piece of state
```suggestion
	local buflist = utils.get_buffers()
	for _, bufnr in ipairs(buflist) do
		state.pinned_buffers[bufnr] = M.pin_all
	end
	M.pin_all = not M.pin_all
	return M.pin_all
```

Reply:
Good point. So, move `state.pin_all` to this module and do some renaming. I'm thinking we should go with `toggle_all` as the name for this function and the boolean value to something more obvious, maybe `M.toggle_all_value`.

Reply:
maybe something like `set_all_to`? The wrapping function performs a toggle indeed, but on the inside what you really do it set all values to some value, so maybe it can reflect that?

Reply:
That sounds good, am I interpreting this correctly?

```lua
local M = {
	set_all_to_value = true,
}

M.set_all_to = function()
	local buflist = utils.get_buffers()
	for _, bufnr in ipairs(buflist) do
		state.pinned_buffers[bufnr] = M.set_all_to_value
	end
	M.set_all_to_value = not M.set_all_to_value
	return M.set_all_to_value
end
```

Reply:
not quite, I was only talking about the variable name, the function name is still a toggle so
```lua
local M = {
	set_all_to_value = true,
}

M.toggle_all = function()
	local buflist = utils.get_buffers()
	for _, bufnr in ipairs(buflist) do
		state.pinned_buffers[bufnr] = M.set_all_to_value
	end
	M.set_all_to_value = not M.set_all_to_value
	return M.set_all_to_value
end
```

Reply:
Now that I looked at both variants I think I like yours better. The one where the function is `toggle_all` and the variable `toggle_all_value` 😅 
Mine is kinda confusing cause it sounds like a method name